### PR TITLE
fix(onlyoffice): #DRIV-24 open doc located in root folder is now working

### PR DESCRIPTION
**Pull request by aymericdevriesere**
## Describe your changes
Fixed url when open file located in nextloud root folder
## Checklist tests
- [x] Open file located in root
- [x] Open file located under a subfolder of root
## Issue ticket number and link
[ DRIV-24 ] 
https://jira.support-ent.fr/browse/DRIV-24
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/services/nextcloud.service.ts [hunk 1]
```diff
@@ -20,8 +20,9 @@ export interface INextcloudService {
 
 export const nextcloudService: INextcloudService = {
     openNextcloudLink: (document: SyncDocument, nextcloudUrl: string): void => {
-        const url: string = document.path;
-        window.open(`${nextcloudUrl}/index.php/apps/files?dir=${url.substring(0, url.lastIndexOf('/'))}&openfile=${document.fileId}`);
+        const url: string = document.path.substring(0, document.path.lastIndexOf('/'));
```
Comment (sjollois) [inline src/main/resources/public/ts/services/nextcloud.service.ts:23]: et si document.path n'a pas de "/" ?

**Review by sjollois — CHANGES_REQUESTED** [review]

**Comment by sjollois** [comment]
il y a une faute sur le titre de ta PR

**Review by BastienBenard — APPROVED** [review]

File: src/main/resources/public/ts/services/nextcloud.service.ts (hunk 1 above)
Comment (aymericdevriesere) [reply src/main/resources/public/ts/services/nextcloud.service.ts:23]: Il en a forcément, les documents dans le folder root sont "/[nom du doc]"

File: src/main/resources/public/ts/services/nextcloud.service.ts [hunk 2]
```diff
@@ -20,8 +20,9 @@ export interface INextcloudService {
 
 export const nextcloudService: INextcloudService = {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/services/nextcloud.service.ts:21]: is* now working (nom du commit/de la PR)

File: src/main/resources/public/ts/services/nextcloud.service.ts (hunk 1 above)
Comment (sjollois) [reply src/main/resources/public/ts/services/nextcloud.service.ts:23]: une petite vérif ne coute rien pour gérer les erreurs et cas particuliers

**Review by alicedraillard — APPROVED** [review]

**Review by BastienBenard — COMMENTED** [review]
C'est possible de faire un test sur cette fonction ?

File: src/main/resources/public/ts/services/nextcloud.service.ts [hunk 3]
```diff
@@ -20,8 +20,9 @@ export interface INextcloudService {
...
     openNextcloudLink: (document: SyncDocument, nextcloudUrl: string): void => {
-        const url: string = document.path;
-        window.open(`${nextcloudUrl}/index.php/apps/files?dir=${url.substring(0, url.lastIndexOf('/'))}&openfile=${document.fileId}`);
+        const url: string = document.path.includes("/") ? document.path.substring(0, document.path.lastIndexOf('/')) : "";             ;
```
Comment (sjollois) [inline src/main/resources/public/ts/services/nextcloud.service.ts:23]: ; en trop

**Review by sjollois — CHANGES_REQUESTED** [review]

**Review by sjollois — APPROVED** [review]

**Pull request merged**